### PR TITLE
feat(core): create rebuild anchor worker

### PR DESCRIPTION
**Pull request by stephhuynh18**
Also locally tested using a real merkle root from QA. I can successfully recreate the anchor commits, and apply them so that they get indexed.

### CODE REVIEW

File: packages/core/src/sync/workers/rebuild-anchor.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  private async getModelForStream(streamId: StreamID): Promise<StreamID | null> {
+    const signedCommit = await this.ipfsService.retrieveCommit(
+      streamId.cid, // genesis commit CID
+      streamId
+    )
+
+    const genesisCommit = await this.ipfsService.retrieveCommit(signedCommit.link, streamId)
+
+    if (!genesisCommit.header.model) {
+      return
```
Comment (PaulLeCam) [inline packages/core/src/sync/workers/rebuild-anchor.ts:32]: Seems this should return `null` here to match the method type definition?

File: packages/core/src/sync/workers/rebuild-anchor.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    const metadata = await merkleTreeLeafLoader.getMetadata().catch(() => {
+      // TODO: add failure job for root cid
+    })
+    if (!metadata) {
+      return
+    }
+
+    const streams = metadata.streamIds
+
+    for (let i = 0; i < streams.length; i++) {
```
Comment (PaulLeCam) [inline packages/core/src/sync/workers/rebuild-anchor.ts:66]: Does this need to be sequential or could this logic run in parallel please?

File: packages/core/package.json [hunk 3]
```diff
@@ -41,6 +41,7 @@
     "clean": "npx rimraf ./lib"
   },
   "dependencies": {
+    "@ceramicnetwork/anchor-utils": "^1.1.0-rc.0",
```
Comment (PaulLeCam) [inline packages/core/package.json:44]: Could it be set in `devDependencies` if the package is only used in tests please?

**Review by PaulLeCam — APPROVED** [review]

File: packages/core/src/sync/workers/rebuild-anchor.ts (hunk 2 above)
Comment (stephhuynh18) [reply packages/core/src/sync/workers/rebuild-anchor.ts]: very right i have changed it to run in parallel thanks!

**Pull request merged**